src/expr/src/linear.rs

```diff
@@ -374,7 +374,10 @@ impl MapFilterProject {
             available_expr[*index] = true;
         }
         for expr in self.expressions.into_iter() {
-            let is_available = expr.support().into_iter().all(|i| available_expr[i]);
+            // We treat an expression as available if its supporting columns are available,
+            // and if it is not a literal (we want to avoid pushing down literals).
```

Comment:
might be worth putting a sentence in as to why we dont want to push down literals